design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,142 @@
+# Ceph COSI Driver Support
+
+## Targeted for v1.12
+
+## Background
+
+Container Object Storage Interface (COSI) is a specification for container orchestration frameworks to manage object storage. Even though there is no standard procotol defined for Object Store, it has flexibility to add support for all. The COSI spec abstracts common storage features such as create/delete buckets, grant/revoke access to buckets, attach/detach buckets, and more. COSI released v1alpha1 with Kubernetes 1.25. COSI is a new project and is not yet fully integrated by Kubernetes.
+More details about COSI can be found [here](https://kubernetes.io/blog/2022/09/02/cosi-kubernetes-object-storage-management/)
+It is projected that COSI will be the only supported object storage driver in the near future. In-tree drivers such as Ceph RGW will be replaced with their respective COSI drivers.
```

Comment:
What in-tree driver would be replaced? Do you mean the OBCs? Anyway, I don't think we will be able to deprecate OBCs for a while unless we have a migration path for them to move to COSI.

Reply:
Agreed. I flagged this same thing in my prior review: https://github.com/rook/rook/pull/11995#discussion_r1180625624